python/ray/_raylet.pyx

```diff
@@ -159,6 +159,22 @@ OPTIMIZED = __OPTIMIZE__
 
 logger = logging.getLogger(__name__)
 
+
+class ObjectRefGenerator:
+    def __init__(self, refs):
+        # TODO(swang): As an optimization, can also store the generator
```

Comment:
Yes, that's possible. The TODO is an optimization for cases where we never use the inner ObjectRefs separately from the outer one. Like if you pass the generator to another task, and that task calls ray.get on each inner ref but deletes the refs before it returns. You can sort of "batch" the ref counts in that case by passing the outer ref to the task instead of the deserialized ObjectRefGenerator.